File: test/integration/shell.rb
#
# Copyright 2011 by Salman Ahmad (salman@salmanahmad.com).
# All rights reserved.
#
# Permission is granted for use, copying, modification, distribution,
# and distribution of modified versions of this work as long as the
# above copyright notice is included.
#

require File.expand_path(File.join(File.dirname(__FILE__), '..', 'test_helper.rb'))

class IntegrationTests::ShellTest < Test::Unit::TestCase
  include RuntimeHelper

  def test_simple
    program = <<-EOD

    DEFINE label FOR shell ON message DO
      PERFORM 'read input; echo input=$input;'
      RETURN images
    END

    PRINT COMPUTE label ON "foo"

    EOD

    tracks, output = run_source(program, true)
    assert_equal('input={"argv":["foo"],"kwargs":{},"name":"label"}', output)

  end

  
end

File: src/main/java/dog/commands/Shell.java

/*
 *
 *  Copyright 2012 by Salman Ahmad (salman@salmanahmad.com).
 *  All rights reserved.
 *
 *  Permission is granted for use, copying, modification, distribution,
 *  and distribution of modified versions of this work as long as the
 *  above copyright notice is included.
 *
 */

package dog.commands;

import dog.util.StringList;
import dog.util.Helper;

import java.util.List;
import java.util.ArrayList;
import java.util.HashMap;

public class Shell extends Command {
	public String description() {
		return "Start a shell session with a running Dog application";
	}

	public void run(StringList args) {
		Helper.eval("<eval>", "forever do;input = console.readLine: \"> \";if input == null then;return;end;dog.print: (dog.eval: input);end");
	}
}





